baseball/src/App.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+  state = {
+    balls: 0,
+    strikes: 0,
+    inning: 1,
+    outs: 0
+  }
+  render() {
+    return (
+      <div className="App">
+        <div className="Display">
+        <h2>Balls: {this.state.balls}</h2>
+        <h2>Strikes: {this.state.strikes}</h2>
+        <h2>Inning: {this.state.inning}</h2>
+        <h2>Outs: {this.state.outs}</h2>
+        </div>
+
+        <div className="Dashboard">
+        <button name="strikes" onClick={this.strike}>Strike</button>
+        <button name="balls" onClick={this.ball}>Ball</button>
+        <button onClick={this.foul}>Foul</button>
+        <button onClick={this.hit}>Hit</button>
+        </div>
+      </div>
+    );
+  }
+
+  handleOuts() {
+    let {outs,inning} = this.state;
```

Comment:
destructuring here is actually great for multiple reasons, not only does it shorten up your references to it later in the code block, but because it's grabbing a reference to the state at the beginning of the function, you can reference it in your `out += 1` case without having to use the `prevState` pattern, since you're not grabbing a_new_ reference to `this.state.outs` inside your setState method :100: